test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,350 @@
+8) Create a custom DNS pod and configure it to use above configMap with agnHost pod's IP.
+9) Create and apply antrea FQDN policy.
+10) Deploy antrea-toolbox.
+
+11) curl the FQDN from within toolbox.
+12) imitate caching the IP belonging to above FQDN resolution by keeping it in a variable.
+13) change the existing IP in dns configmap with the other agnHost pod's IP.
+14) wait for new IP to get updated in configMap and let the changes be reflected in dns pod.
+15) curl the FQDN again with old IP , simulating usage of cache -- and it must fail with no connectivity.
+*/
+
+const (
+	testFullyQualifiedDomainName = "fqdn-test-pod.lfx.test"
+
+	customDnsServiceName = "custom-dns-service"
+	customDnsConfigName  = "custom-dns-config"
+	customDnsPort        = 53
+	customDnsTargetPort  = 53
+
+	customDnsImage         = "coredns/coredns:1.11.3"
+	customDnsPodName       = "custom-dns-server"
+	customDnsContainerName = "coredns"
+	customDnsLabelKey      = "app"
+	customDnsLabelValue    = "custom-dns"
+	customDnsVolume        = "config-volume"
+
+	fqdnPolicyName            = "test-anp-fqdn"
+	fqdnPodSelectorLabelKey   = "app"
+	fqdnPodSelectorLabelValue = "fqdn-cache-test"
+	toolBoxPodName            = "toolbox"
+
+	agnHostPort          = 80
+	agnHostLabelKey      = "app"
+	agnHostLabelValue    = "agnhost"
+	agnHostPodNamePreFix = "agnhost-"
+
+	eventualWaitTime = 20 * time.Second
+)
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
+	skipIfAntreaPolicyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	customDnsService, err := data.CreateServiceWithAnnotations(customDnsServiceName, data.testNamespace, customDnsPort, customDnsTargetPort, corev1.ProtocolUDP,
+		map[string]string{customDnsLabelKey: customDnsLabelValue}, false, false, corev1.ServiceTypeClusterIP, ptr.To[corev1.IPFamily](corev1.IPv4Protocol), map[string]string{})
+	require.NoError(t, err, "Error when creating custom DNS service: %v", err)
+
+	setDnsServerAddressInAntrea(t, data, customDnsService.Spec.ClusterIP)
+
+	podCount := 2
+	agnHostPodIps := make([]*PodIPs, podCount)
+	for i := 0; i < podCount; i++ {
+		agnHostPodIps[i] = createHttpAgnhostPod(t, data, randName(agnHostPodNamePreFix), map[string]string{agnHostLabelKey: agnHostLabelValue})
+	}
+
+	ipv4, ipv6 := extractIPs(agnHostPodIps[0])
+
+	dnsConfigData := createDnsConfig(t, ipv4, ipv6)
+	customDnsConfigMapObject, err := data.CreateConfigMap(data.testNamespace, customDnsConfigName, dnsConfigData, nil, false)
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+
+	createDnsPod(t, data)
+
+	k8sUtils, err = NewKubernetesUtils(data)
+	require.NoError(t, err, "error getting k8s utils %+v", err)
+
+	_ = buildFqdnPolicy(t, data)
+
+	createToolBoxPod(t, data, customDnsService.Spec.ClusterIP)
+
+	curlFqdn := func(podName, containerName, fqdn string) error {
```

Comment:
```suggestion
	curlTarget := func(podName, containerName, target string) error {
```

"fqdn" is a bit misleading here, because we use this function to curl both an fqdn and an IP address